### PR TITLE
MAINT/BUG: Multi-thread logic fix for align_to_tree_mafft_fasttree when n_threads='auto'

### PR DESCRIPTION
**Pull request by lizgehret**
Fixes: #80

### CODE REVIEW

**Review by thermokarst — CHANGES_REQUESTED** [review]
This looks great @lizgehret! I have one note for a little edge-case - please let me know if you have any questions.

File: q2_phylogeny/_fasttree.py [hunk 1]
```diff
@@ -38,8 +38,10 @@ def fasttree(alignment: AlignedDNAFASTAFormat,
         cmd = ['FastTree']
     else:
         env = os.environ.copy()
-        n_threads = 0 if n_threads == 'auto' else n_threads
-        env.update({'OMP_NUM_THREADS': str(n_threads)})
+        if n_threads == 'auto':
+            n_threads = 0
```
[the remark this answers is not in the document]
Comment (thermokarst) [reply q2_phylogeny/_fasttree.py:42]:
```bash
OMP_NUM_THREADS=20 qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences rep-seqs.qza \
  --o-alignment aligned-rep-seqs.qza \
  --o-masked-alignment masked-aligned-rep-seqs.qza \
  --o-tree unrooted-tree.qza \
  --o-rooted-tree rooted-tree.qza \
  --p-n-threads auto \
  --verbose
...
Command: FastTreeMP -quote -nt /var/folders/jt/rdhy5qm11wbgvsfzgbfw_3jw0000gn/T/qiime2-archive-jk5g4kwf/97361a3d-4362-4e02-afd6-378dd13d3aeb/data/aligned-dna-sequences.fasta

FastTree Version 2.1.10 Double precision (No SSE3), OpenMP (20 threads)
...
Saved FeatureData[AlignedSequence] to: aligned-rep-seqs.qza
Saved FeatureData[AlignedSequence] to: masked-aligned-rep-seqs.qza
Saved Phylogeny[Unrooted] to: unrooted-tree.qza
Saved Phylogeny[Rooted] to: rooted-tree.qza
```
<details>
<summary>full output</summary>
```bash
OMP_NUM_THREADS=20 qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences rep-seqs.qza \
  --o-alignment aligned-rep-seqs.qza \
  --o-masked-alignment masked-aligned-rep-seqs.qza \
  --o-tree unrooted-tree.qza \
  --o-rooted-tree rooted-tree.qza \
  --p-n-threads auto \
  --verbose
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: mafft --preservecase --inputorder --thread -1 /var/folders/jt/rdhy5qm11wbgvsfzgbfw_3jw0000gn/T/qiime2-archive-i55z5rkl/9abeef9d-ebde-4052-b9f0-938efe2cd7d1/data/dna-sequences.fasta

OS = darwin
The number of physical cores =  4
inputfile = orig
770 x 120 - 120 d
nthread = 4
nthreadpair = 4
nthreadtb = 4
ppenalty_ex = 0
stacksize: 8192 kb
generating a scoring matrix for nucleotide (dist=200) ... done
Gap Penalty = -1.53, +0.00, +0.00



Making a distance matrix ..
  701 / 770 (thread    2)
done.

Constructing a UPGMA tree (efffree=0) ...
  760 / 770
done.

Progressive alignment 1/2...
STEP   347 / 769 (thread    3) f
Reallocating..done. *alloclen = 1241
STEP   701 / 769 (thread    1) h
done.

Making a distance matrix from msa..
  700 / 770 (thread    1)
done.

Constructing a UPGMA tree (efffree=1) ...
  760 / 770
done.

Progressive alignment 2/2...
STEP   601 / 769 (thread    0) f
Reallocating..done. *alloclen = 1241
STEP   701 / 769 (thread    3) h
done.

disttbfast (nuc) Version 7.475
alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0
4 thread(s)


Strategy:
 FFT-NS-2 (Fast but rough)
 Progressive method (guide trees were built 2 times.)

If unsure which option to use, try 'mafft --auto input > output'.
For more information, see 'mafft --help', 'mafft --man' and the mafft page.

The default gap scoring scheme has been changed in version 7.110 (2013 Oct).
It tends to insert more gaps into gap-rich regions than previous versions.
To disable this change, add the --leavegappyregion option.

Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: FastTreeMP -quote -nt /var/folders/jt/rdhy5qm11wbgvsfzgbfw_3jw0000gn/T/qiime2-archive-jk5g4kwf/97361a3d-4362-4e02-afd6-378dd13d3aeb/data/aligned-dna-sequences.fasta

FastTree Version 2.1.10 Double precision (No SSE3), OpenMP (20 threads)
Alignment: /var/folders/jt/rdhy5qm11wbgvsfzgbfw_3jw0000gn/T/qiime2-archive-jk5g4kwf/97361a3d-4362-4e02-afd6-378dd13d3aeb/data/aligned-dna-sequences.fasta
Nucleotide distances: Jukes-Cantor Joins: balanced Support: SH-like 1000
Search: Normal +NNI +SPR (2 rounds range 10) +ML-NNI opt-each=1
TopHits: 1.00*sqrtN close=default refresh=0.80
ML Model: Jukes-Cantor, CAT approximation with 20 rate categories
Initial topology in 0.16 seconds0 of    748
Refining topology: 38 rounds ME-NNIs, 2 rounds ME-SPRs, 19 rounds ML-NNIs
Total branch-length 38.200 after 0.80 sec, 1 of 749 splits
ML-NNI round 1: LogLk = -23138.198 NNIs 145 max delta 9.74 Time 1.11ges (max delta 9.736)
Switched to using 20 rate categories (CAT approximation)6 of 20
Rate categories were divided by 0.988 so that average rate = 1.0
CAT-based log-likelihoods may not be comparable across runs
Use -gamma for approximate but comparable Gamma(20) log-likelihoods
ML-NNI round 2: LogLk = -19870.700 NNIs 85 max delta 3.38 Time 1.46nges (max delta 3.365)
ML-NNI round 3: LogLk = -19847.179 NNIs 30 max delta 5.19 Time 1.65nges (max delta 5.194)
ML-NNI round 4: LogLk = -19841.825 NNIs 6 max delta 2.16 Time 1.77nges (max delta 0.000)
ML-NNI round 5: LogLk = -19840.739 NNIs 5 max delta 0.36 Time 1.82
ML-NNI round 6: LogLk = -19840.659 NNIs 0 max delta 0.00 Time 1.83
Turning off heuristics for final round of ML NNIs (converged)
ML-NNI round 7: LogLk = -19839.544 NNIs 8 max delta 0.13 Time 2.10 (final)x delta 0.127)
Optimize all lengths: LogLk = -19837.373 Time 2.18
Total time: 2.59 seconds Unique: 751/770 Bad splits: 0/748ternal splits
Saved FeatureData[AlignedSequence] to: aligned-rep-seqs.qza
Saved FeatureData[AlignedSequence] to: masked-aligned-rep-seqs.qza
Saved Phylogeny[Unrooted] to: unrooted-tree.qza
Saved Phylogeny[Rooted] to: rooted-tree.qza
```
</details>

[the remark this answers is not in the document]
Comment (lizgehret) [reply q2_phylogeny/_fasttree.py]: Thanks @thermokarst! I've pushed up modifications we worked on today that account for this edge case. Will work on adding unit tests for this new behavior first thing tomorrow morning.

File: q2_phylogeny/tests/test_fasttree.py [hunk 2]
```diff
@@ -68,6 +68,48 @@ def test_fasttree_n_threads(self):
...
+    #            [-lg] [-wag] [-nt] [-gtr] [-gtrrates ac ag at cg ct gt]
+    #            [-gtrfreq A C G T]
+    #            [ -constraints constraintAlignment
+    #            [ -constraintWeight 100.0 ] ]
+    #            [-log logfile]
+    #          [ alignment_file ]
+    #         [ -out output_newick_file | > newick_tree]
+    fasttree('-expert', n_threads=1)
+    captured = capfd.readouterr()
+    assert 'FastTree' in captured.err
```
Comment (thermokarst) [inline q2_phylogeny/tests/test_fasttree.py:101]: I think this test should be

```suggestion
    assert 'OpenMP' not in captured.err
```

**Review by thermokarst — CHANGES_REQUESTED** [review]
Lookin' good!

**Review by thermokarst — APPROVED** [review]
🦖
